### PR TITLE
Extract few more APIs for w and idp commands

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This just extracts code used in the cmd_*.c files and make nice API out of them, so I can use them in Cutter instead of calling commands.

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/cbin.c [hunk 1]
```diff
@@ -5243,3 +5243,22 @@ RZ_API bool rz_core_bin_archs_print(RzBin *bin, RzCmdStateOutput *state) {
 	rz_cmd_state_output_array_end(state);
 	return true;
 }
+
+RZ_API bool rz_core_bin_pdb_load(RZ_NONNULL RzCore *core, RZ_NONNULL const char *filename) {
```
Comment (XVilka) [inline librz/core/cbin.c:5247]: Could you please avoid any PDB changes until https://github.com/rizinorg/rizin/pull/1517 is merged?

**Review by XVilka — CHANGES_REQUESTED** [review]
The rest looks good

File: librz/core/cbin.c (hunk 1 above)
Comment (ret2libc) [reply librz/core/cbin.c:5247]: If I don't do this, cutter cannot work.

Comment (XVilka) [reply librz/core/cbin.c:5247]: Ok, then. I hope it will not produce any conflicts.

Comment (XVilka) [reply librz/core/cbin.c:5247]: Yes, it should not produce conflicts, it seems: https://github.com/rizinorg/rizin/pull/1517/files#diff-3f97fa549a4b7752e60c646c781a7c013401b28b9ddc18efc0a0a5cc3d713d79

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
```
../librz/core/cio.c
../librz/core/cio.c: In function ‘rz_core_write_base64_at’:
../librz/core/cio.c:822:2: error: ‘buf’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  822 |  free(buf);
      |  ^~~~~~~~~
```

**Comment by ret2libc** [comment]
@XVilka i moved one of your type api, please check if ok

File: librz/core/cio.c [hunk 2]
```diff
@@ -648,3 +648,213 @@ RZ_API RzCmdStatus rz_core_io_plugins_print(RzIO *io, RzCmdStateOutput *state) {
...
+ *
+ * The values read/written are considered as integers of \p sz bytes.
+ *
+ * \param core RzCore reference
+ * \param addr Address where to overwrite the value
+ * \param value Value to sum to the existing value in \p addr
+ * \param sz Size of the values, in bytes, to consider. Can be 1, 2, 4, 8.
+ */
+RZ_API bool rz_core_write_value_inc_at(RzCore *core, ut64 addr, st64 value, int sz) {
+	rz_return_val_if_fail(sz == 1 || sz == 2 || sz == 4 || sz == 8, false);
```
Comment (XVilka) [inline librz/core/cio.c:712]: What about using the same macro in the previous function as well?

File: librz/core/cio.c [hunk 3]
```diff
@@ -648,3 +648,213 @@ RZ_API RzCmdStatus rz_core_io_plugins_print(RzIO *io, RzCmdStateOutput *state) {
...
+}
+
+/**
+ * \brief Write a given string \p s at the specified \p addr encoded as base64.
+ *
+ * \param core RzCore reference
+ * \param addr Address where to write the string
+ * \param s String to encode as base64 and then written.
+ */
+RZ_API bool rz_core_write_base64_at(RzCore *core, ut64 addr, const char *s) {
```
Comment (XVilka) [inline librz/core/cio.c:795]: Check if `s` is NULL, add `RZ_NONNULL`

File: librz/core/cio.c [hunk 4]
```diff
@@ -648,3 +648,213 @@ RZ_API RzCmdStatus rz_core_io_plugins_print(RzIO *io, RzCmdStateOutput *state) {
...
+}
+
+/**
+ * \brief Write a given base64 string \p s at the specified \p addr, decoded
+ *
+ * \param core RzCore reference
+ * \param addr Address where to write the string
+ * \param s String to decode from base64 and then written
+ */
+RZ_API bool rz_core_write_base64d_at(RzCore *core, ut64 addr, const char *s) {
```
Comment (XVilka) [inline librz/core/cio.c:844]: Same here

**Review by XVilka — CHANGES_REQUESTED** [review]
Yes, the typelink API change is fine since it's temporary anyway, as you noted due to the future migration to the global variables.

**Review by XVilka — APPROVED** [review]

**Pull request merged**